vllm/v1/engine/core.py

```diff
@@ -370,6 +364,31 @@ def run_busy_loop(self):
             if outputs is not None:
                 self.output_queue.put_nowait(outputs)
 
+    def step(self) -> EngineCoreOutputs:
+        """Schedule, execute, and make output."""
+
+        # Check for any requests remaining in the scheduler - unfinished,
+        # or finished and not yet removed from the batch.
+        if not self.scheduler.has_requests():
+            return EngineCoreOutputs(
+                outputs=[],
+                scheduler_stats=self.scheduler.make_stats(),
+            )
+        scheduler_output = self.scheduler.schedule()
+        output = self.model_executor.execute_model(scheduler_output)
+
+        # Process new inputs here to avoid creating outputs for aborted
+        # requests and to reflect new requests in waiting queue stats.
+        self._process_input_queue_nonblock()
```

Comment:
This reduces messages sent when e.g. stop strings are used. The front-end will have sent an abort during the forward pass - if we process it here we won't create a useless response for the aborted request in `update_from_output` below.